launcher/sdw_updater_gui/UpdaterApp.py

```diff
@@ -0,0 +1,256 @@
+        self.background_update_thread.signal.connect(self.update_status)
+
+    # This slot will receive update signals from UpdateThread
+    def update_status(self, result):
+        self.logOutputText += result["output"]
+        self.logOutput.setText(self.logOutputText)
+        self.progress += 15
+        self.progressBar.setProperty("value", self.progress)
+
+        if result["run_completed"]:
+            if result["action"] == "update":
+                updates_required = False
+                self.progressBar.setProperty("value", 100)
+                for vm in VMS:
+                    if result["{}_update_required".format(vm)]:
+                        updates_required = True
+                        self.vms_to_update.append(vm)
+                if updates_required:
+                    self.proposedActionLabel.setText(
+                        "Updates are available and must be completed before launching client."
+                    )
+                    # write a wile to sd-svs, for client to parse
+                    try:
+                        subprocess.check_call(
+                            ["qvm-run", "sd-svs", "touch {}".format(UPDATE_FLAGFILE)]
+                        )
+                        self.applyUpdatesButton.setEnabled(True)
+                    except subprocess.CalledProcessError as e:
+                        self.proposedActionLabel.setText(
+                            "Error writing file to sd-svs."
+                        )
+                        self.logOutputText += "Failed to write file to sd-svs, please contact your administrator"
+                        self.logOutput.setText(self.logOutputText)
+                else:
+                    self.proposedActionLabel.setText("System up-to-date.")
+                    # remove the flag file to indicate to sd-svs that critical VMs are up-to-date
+                    try:
+                        subprocess.check_call(
+                            ["qvm-run", "sd-svs", "rm -f {}".format(UPDATE_FLAGFILE)]
+                        )
+                        self.clientOpenButton.setEnabled(True)
+                    except subprocess.CalledProcessError as e:
+                        self.proposedActionLabel.setText(
+                            "Error writing file to sd-svs."
+                        )
+                        self.logOutputText += "Failed to write file to sd-svs, please contact your administrator"
+                        self.logOutput.setText(self.logOutputText)
+            elif result["action"] == "upgrade":
+                self.proposedActionLabel.setText(
+                    "Updates have been applied, system is now up-to-date."
+                    # TODO: Reboot if xen updates, don't remove flag for now
+                )
+                self.progressBar.setProperty("value", 100)
+                self.clientOpenButton.setEnabled(True)
+
+        self.logOutputText += str(result)
```

Comment:
Nit: after each write to `self.logOutputText` we need a line break otherwise all the output is concatenated (if you want you could add a helper method on UpdaterApp takes a string, adds a line break, and calls `self.logOutput.setText`)